src/App.jsx

```diff
@@ -1,12 +1,19 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import reactLogo from './assets/react.svg'
 import viteLogo from '/vite.svg'
 import './App.css'
 import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
+function useSemiPersistentState(key, value) {
+  let [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")) || []);
```

Comment:
You can use `const` instead of `let` since those variables won't get reassigned.

---

It's preferable to pass an initializer function as argument to `useState` when the initial value requires any expensive computation. Reading from local storage and parsing a string to JSON can be both considered expensive, so running them only once instead of on every render will improve the performance.

```js
const [todoList, setTodoList] = useState(() => JSON.parse(localStorage.getItem("savedTodoList")) || []);
```

The reason is that every time React needs to render the page, it will run the component function. Even though the initial state is already set, the instructions to parse and read from the storage are there and will run again. When you pass a function to `useState`, React will make sure of running it only once.